packages/react/react/src/hooks/useStorageState.ts

```diff
@@ -2,36 +2,35 @@
 import { safeLocalStorage, Storage } from '@toss/storage';
 import { SetStateAction, useCallback, useState } from 'react';
 
-export type Serializable<T> = T extends string | number | boolean | unknown[] | Record<string, unknown> ? T : never;
-
+export type Serializable = string | number | boolean | unknown[] | Record<string, unknown>;
 interface StorageStateOptions<T> {
   storage?: Storage;
-  defaultValue?: Serializable<T>;
+  defaultValue?: T;
 }
 
 interface StorageStateOptionsWithDefaultValue<T> extends StorageStateOptions<T> {
-  defaultValue: Serializable<T>;
+  defaultValue: T;
 }
 
 /**
  * 스토리지에 상태값을 저장하고 불러와서 값이 보존되는 useState로 동작하는 hook입니다.
  * @param key 스토리지에 저장할 키
  */
-export function useStorageState<T>(
+export function useStorageState<T extends Serializable>(
   key: string
-): readonly [Serializable<T> | undefined, (value: SetStateAction<Serializable<T> | undefined>) => void, () => void];
+): readonly [T | undefined, (value: SetStateAction<T | undefined>) => void, () => void];
 export function useStorageState<T>(
   key: string,
   { storage, defaultValue }: StorageStateOptionsWithDefaultValue<T>
-): readonly [Serializable<T>, (value: SetStateAction<Serializable<T>>) => void, () => void];
+): readonly [T, (value: SetStateAction<T>) => void, () => void];
 export function useStorageState<T>(
   key: string,
   { storage, defaultValue }: StorageStateOptions<T>
-): readonly [Serializable<T> | undefined, (value: SetStateAction<Serializable<T> | undefined>) => void, () => void];
+): readonly [T | undefined, (value: SetStateAction<T | undefined>) => void, () => void];
 export function useStorageState<T>(
   key: string,
   { storage = safeLocalStorage, defaultValue }: StorageStateOptions<T> = {}
-): readonly [Serializable<T> | undefined, (value: SetStateAction<Serializable<T> | undefined>) => void, () => void] {
+): readonly [T | undefined, (value: SetStateAction<T | undefined>) => void, () => void] {
```

Comment:
@Jinho1011 
First of all, thank you for your interest in my PR.

Is there a reason for putting a generic constraint only in the first function definition? Could it be a mistake?

Reply:
Thank you for letting me know. I fixed the missing one and pushed a new commit with the changes.